wfe/wfe.go

```diff
@@ -1100,7 +1100,14 @@ func (wfe *WebFrontEndImpl) Registration(ctx context.Context, logEvent *requestE
 		return
 	}
 
-	if len(update.Agreement) > 0 && update.Agreement != wfe.SubscriberAgreementURL {
+	// If a user POSTs their registration object including a previously valid
+	// agreement URL but that URL has since changed we will fail out here
+	// since the update agreement URL doesn't match the current URL. To fix that we
```

Comment:
Sorry to nitpick the comment, but: This still doesn't explain *why* we do this check here rather than in the RA. It should explain that "no diff" updates are short-circuited in the RA, but the WFE performs an early check for subscriber agreement URL, so we need an extra short-circuit here.